napari/_qt/qt_main_window.py

```diff
@@ -492,6 +501,24 @@ def _help_changed(self, event):
         """
         self._help.setText(event.text)
 
+    def _save_screenshot(self):
```

Comment:
i find it a little odd that we now have `screenshot` and `_save_screenshot`, where `_save_screenshot` doesn't actually do the saving, it just generates a dialog window, and `screenshot` does the saving, but only with the `path=` argument...

Reply:
This pattern exists in `QtViewer` too - I am happy to change it, and now is a fine time to do so, but then we should do it in both

Reply:
Yeah I find it a bit weird also. I even wonder if we should remove it from the public API, and make it a private utility with menubar access. We should start to be careful with our growing public API footprint, lest our viewer become a wasteland like the numpy main namespace.

Reply:
ok - will go with `_save_screenshot` -> `_screenshot_dialog`. I think having `screenshot` as a public api makes sense. Note if `path=None` nothing will get saved and in either case you'll get back the actual screenshot